### PR TITLE
feat(pie-docs): DSW-2268 refactors navigation to include link to inde…

### PR DESCRIPTION
**Pull request by maledr5**
## Describe your changes (can list changeset entries if preferable)
Refactor the navigation mechanism so that:
- main navigation items can link to index pages
- caret can independently open the accordion without linking

Decided to keep the focus navigation that will select the `summary` element as well as the `link` and `button`. Is one additional tab step but I did not wanted to remove the a11y advantages of using `<details>` and `<summary>` elements.

**Note on percy changes:** 
- The menu was moved 2px down so all screenshots have changed.
- I have left an [empty index page for Components](https://pr1708-docs.pie.design/components/) so to showcase how it will work if one of the main menu items has content. I will be working on the index page content next so this navigation and the content changes can be merged together. But I wanted to get a review on this changes first.

**Note on resizeObserver removal:**
There is no record of the decision on why we implemented the animation only for desktop and not mobile screens. As this solution relies on the initial animation script, I removed the check for smaller screens (>=600). Without this check I could not see a use for the resizeObserver either, I wasn't able to understand why we would need it even before as we set the accordions once and do not change them on resize any more. If anybody has more context let me know. 

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the Docs site preview
- [x] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1 - @jamieomaguire 
- [x] I have reviewed the Docs site preview
- [x] I have reviewed the new navigation interaction
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2
- [x] I have reviewed the Docs site preview
- [x] I have reviewed the new navigation interaction
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/assets/js/accordions.js [hunk 1]
```diff
@@ -122,27 +128,11 @@ class Accordion {
...
-
-            // ensure we don't duplicate accordion instances
-            if (width > 600 && !accordions.length) {
-            // TODO: use fozzie JS breakpoint helpers instead of hardcoding 600px
-                document.querySelectorAll('.c-nav details').forEach((el) => {
-                    accordions.push(new Accordion(el));
-                });
-            }
-        });
-    });
```
Comment (raoufswe) [inline apps/pie-docs/src/assets/js/accordions.js:139]: we don't need this logic anymore?

File: apps/pie-docs/src/_includes/nav.njk [hunk 2]
```diff
@@ -50,16 +66,17 @@
...
+                            {{ summary(category.title, category.url, category.children[0].url, category.hasIndexPage, category.key) }}
                             {% if category.children.length %}
                             {# build up a list of subCategories for each subpages (used in component pages) #}
                                 <ul class="c-nav-list">
                                     {% for subpage in category.children %}
                                         {% if subpage.url %}
                                             {% if subpage.children.length and subpage.subPageDropdown %}
-                                                <details {% if page.url and page.url.includes(subpage.key) %} open{% endif %}>
-                                                    {{ summary(subpage.title) }}
+                                                <details>
```
Comment (raoufswe) [inline apps/pie-docs/src/_includes/nav.njk:78]: same as above

File: apps/pie-docs/src/assets/js/accordions.js (hunk 1 above)
Comment (maledr5) [reply apps/pie-docs/src/assets/js/accordions.js:139]: Thanks @raoufswe I left a comment around this in the description :) , here it is:

There is no record of the decision on why we implemented the animation only for desktop and not mobile screens. As this solution relies on the initial animation script, I removed the check for smaller screens (>=600). Without this check I could not see a use for the resizeObserver either. I wasn't able to understand why we would need it before as we set the accordions once and do not change them on resize any more. If anybody has more context let me know.

Do you know why we need it or remember context around this decision?

File: apps/pie-docs/src/_includes/nav.njk [hunk 3]
```diff
@@ -50,16 +66,17 @@
             {% for category in section %}
                 {% if category.url %}
                     <li>
-                        <details {% if page.url and page.url.includes(category.url) %} open{% endif %}>
-                            {{ summary(category.title, category.key) }}
+                        <details>
```
[the remark this answers is not in the document]
Comment (maledr5) [reply apps/pie-docs/src/_includes/nav.njk:69]: The issue here is that now we navigate. The moment we navigate we lose the animation. So what I do now is trigger the animation after the navigation happens (after the re-render). if I add this line again, as it is a default html function that will make the list visible, with the current solution creates this weird jump and of course removes the animation.

The only solution I can think about is remembering the previous state using browser local storage: storing the previous url and only triggering the animation when it changes. Any thoughts?

https://github.com/user-attachments/assets/810088f7-1608-444c-a30e-0d4776516d14

[the remark this answers is not in the document]
Comment (maledr5) [reply apps/pie-docs/src/_includes/nav.njk]: Added this line back and removed animation when clicking on links as agreed with the team :)

File: apps/pie-docs/src/_includes/nav.njk (hunk 2 above)
Comment (maledr5) [reply apps/pie-docs/src/_includes/nav.njk]: Added this line back and removed animation when clicking on links as agreed with the team :)

File: apps/pie-docs/src/assets/js/accordions.js (hunk 1 above)
Comment (raoufswe) [reply apps/pie-docs/src/assets/js/accordions.js:139]: maybe @jamieomaguire has any idea? 👀

**Review by raoufswe — DISMISSED** [review]

File: apps/pie-docs/src/assets/js/accordions.js (hunk 1 above)
Comment (jamieomaguire) [reply apps/pie-docs/src/assets/js/accordions.js:139]: Ash asked for it, I delivered :P 

In all seriousness, it was something we felt was a nice UX improvement. Why we left out mobile, I cannot remember unfortunately. 

I think it is perfectly fine for us to remove this for now and revisit after chatting with design :)

**Review by jamieomaguire — APPROVED** [review]

**Review by jamieomaguire — APPROVED** [review]

**Review by raoufswe — APPROVED** [review]

**Pull request merged**